examples/camera-controller/webrtc_manager/WebRTCProviderClient.h

```diff
@@ -119,6 +129,17 @@ class WebRTCProviderClient : public chip::app::CommandSender::Callback
         kEndSession           = 5,
     };
 
+    enum class State : uint8_t
+    {
+        Idle,             ///< Default state, no communication initiated yet
+        Connecting,       ///< Waiting for OnDeviceConnected or OnDeviceConnectionFailure callbacks to be called
+        AwaitingResponse, ///< Waiting for command response from camera
+        AwaitingAnswer,   ///< Waiting for Answer command from camera
+    };
```

Comment:
There should be a Closed state to which the FSM transitions to after the successful EndSession flow. On the controller side, it sends the EndSession and awaits the response before transitioning to Closed state.
That could be a subsequent update to the FSM in a later PR?

Reply:
What is the difference between Closed state and Idle State? We finally needs to go back to the Idle state after enter Closed state, what is the trigger?

Reply:
The Closed state could actually be made the same as Idle as it is a pass-thorugh state to Idle. However, then we might need a transient EndSessionSent state where the local entity has closed its side of the session. 
If we collapse all intermediate states to Idle in order to simplify the FSM, then it may not be clear what metadata the device should and should not have when in Idle state(Session, Ice candidates, etc).

Reply:
Per discussion, it is very tricky to use state to handle ICECandidates exchange, right now, we just use FSM to guard SDP exchange, we can add additional state to handle ICECandidates exchange later